### ISSUE TITLE
Desenvolvimento do Site [Roadmap]

### ISSUE THREAD

**Issue by ktquez** [svelte-brasil/site#1]
Fala pessoal, blz

Estamos nas etapas iniciais para o desenvolvimento do site/blog Svelte Brasil, gostaria de convidar quem quiser participar.

Essa issue é para começarmos a decisão de onde e como começar, quais tcnologias iremos usar e etc.

O layout é possível de ver nesse link:
https://www.behance.net/gallery/80546401/Svelte-Brasil

**Comment by JonasJs** [comment]
Podemos usar o Firebase como banco de dados ele aguenta bastante dados, Assim conseguimos hospedar no github pages de boas.
Isso tudo Free rs

**Comment by p-moreira** [comment]
Minha  opinião, de acordo com o que já foi discutido no grupo:

- Usar o [Sapper](https://sapper.svelte.dev)
- Decidir se o site vai usar SSR ou vai ser de páginas estáticas. As duas estratégias são boas pra SEO, lembrando que SSR precisa de um server pago.

**Comment by ktquez** [comment]
Sou a favor de ser estático.

**Comment by JonasJs** [comment]
Também acho que no primeiro momento pode ser "estatico" e logo depois iniciamos uma nova etapa do projeto

**Comment by p-moreira** [comment]
Sendo estático e sem custo, acho que melhor stack pra começar e simplificar seria:
- sapper
- netlify
- netlify CMS

**Comment by JonasJs** [comment]
@p-moreira Tava pensando em adicionar o Firebase que é um back end pronto, só precisamos enviar as Request pros endpoints, tudo no front, assim conseguimos ter esse dados gravador, alem de conseguir adicionar novas features que ajude os leitores.

e logo depois que estiver tudo certo, podemos iniciar um projeto maior com um back end próprio, pois ja teremos mais colaboradores na comunidade para ajudar o projeto ainda mais.

**Comment by ktquez** [comment]
Esse final de semana pessoal a gente se junta e fatia o projeto em issues para gerenciarmos os cards aqui:
https://github.com/svelte-brasil/site/projects/1

**Comment by p-moreira** [comment]
@JonasJs , netlify CMS vai lidar com algumas coisas chatinhas que teríamos que fazer do zero usando o Firebase. Ele já entrega uma interface pronta para edição dos posts. Usando o Firebase teríamos que construir uma área administrativa (um CMS próprio). Considere isso.

**Comment by JonasJs** [comment]
@p-moreira aaaaaa Ai Sim, não o conhecia

**Comment by ktquez** [comment]
Nada pessoal, usamos o Sapper tudo estático, faremos algo que leia os arquivos de markdown + YAML frontmatter e geramos as páginas em HTML no generate.

Os posts poderão ser enviados via PR + Post review, daí a gente coloca alguma ferramenta de CI para quenao for aprovado o PR, ele gere o novo build e atualize o site automaticamente (de preferência usando o Github pages).

Eu estou pensando em um fluxo assim, o que acham

**Comment by ktquez** [comment]
## Desenvolvimento
- HTML semântico e acessível
- Tailwind CSS (https://tailwindcss.com/docs)
- Svelte (https://svelte.dev/docs)


## Testes
- Jest (https://github.com/ktsn/svelte-jest)
- Cypress


## CI & CD
- https://circleci.com/

## QA
- https://deepscan.io/

**Comment by JonasJs** [comment]
@ktquez o Layout foi baseado no Tailwind CSS ?

Shoooow Vamoos começae 😍

**Comment by p-moreira** [comment]
@ktquez  apenas dois pontos que eu não concordo:

1. Não confio no GitHub Pages e deixaria os arquivos em um serviço que já cuida de deploy automático e é mais confiável como o [netlify](https://www.netlify.com/) ou o [now](https://zeit.co/now). 

2. Usar uma interface gráfica pra escrever os posts, melhora muito a experiência do usuário. Usar um editor de código, apesar de ser válido, não é uma solução com foco no usuário, mesmo que o usuário seja você.

**Comment by ktquez** [comment]
@p-moreira 

Por que você não confia no `GitHub Pages`?

Se a preferência for para ter uma GUI, então é melhor usarmos o netlify CMS mesmo e hospedar na Netlify.

Fechado então, montaremos um Svelte Static Site Generator e configuramos um plugin para o Netlify CMS.

**Comment by p-moreira** [comment]
@ktquez , depoimentos de colegas que tinham projetos iniciantes no Pages e estavam tendo problemas com delay no acesso (discussão num outro grupo), como se os servidores dormissem. Eles mudaram pra netlify e notaram a diferença no carregamento das páginas e os acessos começaram a aumentar no analytics. ([cssscan](https://getcssscan.com/)) é um produto de Guilherme (brasileiro), que passou por isso.

O netlify CMS grava direto no github. Então a sua ideia inicial continua a mesma de gerar os arquivos estáticos a partir do markdown gravado do github.

**Comment by patrickalima98** [comment]
Fala galera, então sobre o site eu andei pesquisando e vendo a opinião de cada um, e gostaria de sugerir o seguinte:

- Primeiro sou a favor do site estático também, acredito que isso mantei simples nossa missão de levar conteudo para o pessoal e facilitar o nosso trabalho :)
- Segundo, também acho melhor criarmos um sistema para ler os arquivos estáticos, entretanto acho que seria interessante ter uma UI para escrever os arrtigos, incluir as imagens e etc... Neste caso sei que tem alguns editores opensource que podemos usar algum editor em JS baseado no Medium por exemplo, vou dar uma procurada nos melhores e ver se podemos usar algum.

o que acham? Com isso poderiamos fazer textos em markdown e via UI e depois converte-los para markdown

**Comment by salatielq** [comment]
Olá, @ktquez gostaria de saber como posso contribuir com o desenvolvimento do site :)

**Comment by rof20004** [comment]
Minha opinião é ser estático e utilizar o netlify.

**Comment by wkoch** [comment]
Alguém aqui já pensou em simplesmente fazer um fork do Svelte.dev e traduzir tudo e adaptar a partir disso?

Achei vocês pq estava cogitando traduzir o site Svelte.dev.

**Comment by rof20004** [comment]
@wkoch Na verdade o objetivo incial é fazer um blog/site, diferente do site oficial mesmo, mas é uma ideia válida incluir o blog no site oficial. Precisamos decidir sobre isso.

**Comment by rof20004** [comment]
@wkoch Se bem que o site oficial já tem blog, dai já não sei se irá conflitar.

**Comment by rof20004** [comment]
@wkoch Exatamente, como não foi eu que dei a ideia, to só apoiando a galera, não sei qual o objetivo mesmo, estou até hoje no aguardo na verdade, hehehe.

**Comment by rof20004** [comment]
@wkoch Mas vc já viu o layout proposto? Ta muito legal e moderno.

**Comment by rof20004** [comment]
@wkoch Eu prefiro o layout proposto pela nossa equipe do que o oficial, fora que será totalmente brasileiro, acho legal a ideia.

**Comment by rof20004** [comment]
@wkoch O @ktquez pode te responder melhor.
